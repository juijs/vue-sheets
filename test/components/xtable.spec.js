import { mount } from 'vue-test-utils'
import SheetsXTable from '@/components/xtable'

describe('xtable.js', () => {
    const wrapper = mount(SheetsXTable, {
        propsData: {
            columns: [
                { key: "name", name: "이름", width: "200px" },
                { key: "age", name: "나이", sort: true },
                { key: "location", name: "지역", sort: true },
            ],
            data: [
                { name: "홍재석", age: 31, location: "대한민국" },
                { name: "홍재석", age: 22, location: "대한민국2" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" },
                { name: "홍재석", age: 12, location: "대한민국3" }
            ],
            scrollHeight: 100,
            templateRow: `<tr>
                <td><!= name !></td>
                <td><!= age !></td>
                <td><!= location !></td>
            </tr>`,
            templateNone: `<tr>
                <td class="none" colspan="3">
                    <div class="msg">None!!!</div>
                </td>
            </tr>`
        }
    });

    it('renders li for each item in props.items', () => {
        expect(wrapper.vm.data.length).toBe(9);
    })

    it('server render', () => {
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
})