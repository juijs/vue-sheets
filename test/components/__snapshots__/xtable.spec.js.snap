// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xtable.js server render 1`] = `
<div
  class="xtable nowrap scroll"
>
  <div
    class="head"
  >
    <table
      class="table classic normal hover"
      style="position: relative;"
    >
      <thead>
        <tr>
          <th
            style="width: 0px;"
            title="이름"
            width="200px"
          >
            이름
          </th>
          <th
            style="cursor: pointer; width: 0px;"
            title="나이"
            width="auto"
          >
            나이
          </th>
          <th
            style="cursor: pointer;"
            title="지역"
            width="auto"
          >
            지역
          </th>
        </tr>
      </thead>
       
    </table>
  </div>
  <div
    class="body"
    style="max-height: 100px; overflow-y: scroll;"
  >
    <div
      style="height: 234px;"
    >
      <table
        class="table classic normal hover"
        style="position: relative; top: 0px;"
      >
        <thead>
          <tr
            style="height: 0px;"
          >
            <th
              style="height: 0px; width: 0px;"
              title="이름"
              width="200px"
            />
            <th
              style="cursor: pointer; height: 0px; width: 0px;"
              title="나이"
              width="auto"
            />
            <th
              style="cursor: pointer; height: 0px; width: auto;"
              title="지역"
              width="auto"
            />
          </tr>
        </thead>
         
        <tbody>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              31
            </td>
            
                
            <td>
              대한민국
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              22
            </td>
            
                
            <td>
              대한민국2
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
          <tr>
            
                
            <td>
              홍재석
            </td>
            
                
            <td>
              12
            </td>
            
                
            <td>
              대한민국3
            </td>
            
            
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
